go/signedexchange/structuredheader/parser.go

```diff
@@ -0,0 +1,259 @@
+// Package structuredheader parses Structured Headers for HTTP
+// (https://httpwg.org/http-extensions/draft-ietf-httpbis-header-structure.html).
+package structuredheader
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// http://httpwg.org/http-extensions/draft-ietf-httpbis-header-structure.html#identifier
+type Identifier string
+
+// http://httpwg.org/http-extensions/draft-ietf-httpbis-header-structure.html#param
+type ParameterisedList []*ParameterisedIdentifier
+type ParameterisedIdentifier struct {
+	Label  Identifier
+	Params Parameters
+}
+
+// Parameter represents a set of parameters in a Parameterised Identifier.
+// The interface{} value is one of these:
+//
+//	int64      for Numbers
+//	string     for Strings
+//	Identifier for Identifiers
+//	[]byte     for Byte Sequences
+//	nil        for parameters with no value
+type Parameters map[Identifier]interface{}
+
+type parser struct {
+	input string
+}
+
+func (p *parser) discardLeadingOWS() {
+	p.input = strings.TrimLeft(p.input, " \t")
```

Comment:
I thought OWS is defined as a space char or a tab char that might be followed by whitespace characters: https://tools.ietf.org/html/rfc7230#section-3.2.3 In this implementation, only tab characters on left side would be removed.

Reply:
Sorry this was my misunderstanding